src/OutputWriters/netcdf_output_writer.jl

```diff
@@ -59,138 +56,144 @@ function write_grid(model; filename="./grid.nc", mode="c",
-                            "v" => Dict("longname" => "Velocity in the y-direction", "units" => "m/s"),
-                            "w" => Dict("longname" => "Velocity in the z-direction", "units" => "m/s"),
-                            "T" => Dict("longname" => "Temperature", "units" => "K"),
-                            "S" => Dict("longname" => "Salinity", "units" => "g/kg"))
+    # Creates an unliimited dimension "time"
+    defDim(dataset, "time", Inf)
+    defVar(dataset, "time", Float64, ("time",))
+    sync(dataset)
+
+    # Ensure we have an attribute for every output. Use reasonable defaults if
+    # none were specified by the user.
+    for c in keys(outputs)
+        if !haskey(output_attributes, c)
+            output_attributes[c] = default_output_attributes[c]
+        end
     end
 
-    # Initiates empty Float32 arrays for fields from the user-supplied variable outputs
-    for (fieldname, field) in outputs
-        dtype = eltype(interiorparent(field))
-        defVar(dataset, fieldname, dtype, (netcdf_spatial_dimensions(field)...,"Time"),
-               compression=compression, attrib=output_attributes[fieldname])
+    # Initiates empty variables for fields from the user-supplied variable outputs
+    for (name, output) in outputs
+        if output isa Field
+            FT = eltype(output.grid)
+            defVar(dataset, name, FT, (netcdf_spatial_dimensions(output)..., "time"),
+                   compression=compression, attrib=output_attributes[name])
+        else
+            defVar(dataset, name, Float64, (dimensions[name]..., "time"),
+                   compression=compression, attrib=output_attributes[name])
+        end
     end
     sync(dataset)
 
     # Stores slices for the dimensions of each output field
     slices = Dict{String, Vector{Union{OrdinalRange,Colon}}}()
-    for (fieldname, field) in outputs
-        slices[fieldname] = slice(field; slice_kw...)
+    for (name, output) in outputs
+        if output isa Field
+            slices[name] = slice(output; slice_kw...)
+        end
     end
 
-    return NetCDFOutputWriter(filename, dataset, outputs, interval,
-                          frequency, clobber, slices, len_time_dimension, 0.0)
+    return NetCDFOutputWriter(filename, dataset, outputs, interval, frequency,
+                              clobber, slices, 0.0)
 end
 
-
 # Closes the outputwriter
-Base.close(fw::NetCDFOutputWriter) = close(fw.dataset)
-
-function read_output(fw::NetCDFOutputWriter, fieldname)
-    ds = Dataset(fw.filename,"r")
-    field = ds[fieldname][:,:,:,end]
-    close(ds)
-    return field
-end
+Base.close(ow::NetCDFOutputWriter) = close(ow.dataset)
 
 """
     slice(field; slice_kw...)
 
-For internal use only. Returns a slice for a field based on its dimensions and the supplied slices in `slice_kw`.
+For internal use only. Returns a slice for a field based on its dimensions and the
```

Comment:
Do we have to say "For internal use only"?

Reply:
True. Not exporting it implies it's mainly for internal use.

Reply:
Let the user decide!